src/test/java/org/openstreetmap/atlas/geography/atlas/SubAtlasTest.java

```diff
@@ -250,14 +250,138 @@ public void testSubAtlasPredicateHardCutRelationsOnly()
+        Assert.assertEquals(1, identicalSubAtlas.relation(5).members().size());
+
+        // Nodes
+        Assert.assertNotNull(source.node(1));
+        // Node 1 gets pulled in by Edge 0
+        Assert.assertNotNull(subAtlasWithZeroBasedIdentifiers.node(1));
+        Assert.assertNotNull(source.node(2));
+        // Node 2 gets pulled in by Edge 0
+        Assert.assertNotNull(subAtlasWithZeroBasedIdentifiers.node(2));
+        Assert.assertNotNull(source.node(3));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.node(3));
+
+        // Edges
+        Assert.assertNotNull(source.edge(0));
+        Assert.assertNotNull(subAtlasWithZeroBasedIdentifiers.edge(0));
+        Assert.assertNotNull(source.edge(1));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.edge(1));
+
+        // Areas
+        Assert.assertNotNull(source.area(0));
+        Assert.assertNotNull(subAtlasWithZeroBasedIdentifiers.area(0));
+        Assert.assertNotNull(source.area(1));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.area(1));
+
+        // Lines
+        Assert.assertNotNull(source.line(0));
+        Assert.assertNotNull(subAtlasWithZeroBasedIdentifiers.line(0));
+        Assert.assertNotNull(source.line(1));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.line(1));
+
+        // Points
+        Assert.assertNotNull(source.point(0));
+        Assert.assertNotNull(subAtlasWithZeroBasedIdentifiers.point(0));
+        // Point1 gets pulled in because it is part of the geometry of line0
+        Assert.assertNotNull(source.point(1));
+        Assert.assertNotNull(subAtlasWithZeroBasedIdentifiers.point(1));
+        Assert.assertNotNull(source.point(2));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.point(2));
+        Assert.assertNotNull(source.point(3));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.point(3));
+
+        // Relations
+        Assert.assertNotNull(source.relation(1));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.relation(1));
+        Assert.assertNotNull(source.relation(2));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.relation(2));
+        Assert.assertNotNull(source.relation(3));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.relation(3));
+        Assert.assertNotNull(source.relation(4));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.relation(4));
+        Assert.assertNotNull(source.relation(5));
+        Assert.assertNull(subAtlasWithZeroBasedIdentifiers.relation(5));
+    }
+
     @Test
     public void testSubAtlasPredicateSoftCut()
     {
         final Atlas source = this.rule.getAtlas();
 
         // Should return back all entities in this atlas
-        final Predicate<AtlasEntity> allEntities = (entity) -> entity.getIdentifier() > 0
-                || entity.getIdentifier() < 0 || entity.getIdentifier() == 0;
+        final Predicate<AtlasEntity> allEntities = entity -> true;
```

Comment:
Nice simplification :)

Reply:
What was the reasoning for the previous way?

Reply:
not sure! legacy from a previous test that I copied and modified.

Reply:
Maybe the coffee machine was out that morning.